pkg/app/piped/executor/kubernetes/kubernetes.go

```diff
@@ -605,3 +606,122 @@ func annotateConfigHashToDeployment(m provider.Manifest, managedConfigMaps, mana
 	)
 	return nil
 }
+
+type patcher func(m provider.Manifest, cfg config.K8sResourcePatch) (*provider.Manifest, error)
+
+func patchManifests(manifests []provider.Manifest, patches []config.K8sResourcePatch, patcher patcher) ([]provider.Manifest, error) {
+	if len(patches) == 0 {
+		return manifests, nil
+	}
+
+	out := make([]provider.Manifest, len(manifests))
+	copy(out, manifests)
+
+	for _, p := range patches {
+		var target = -1
+		for i, m := range out {
+			if m.Key.Kind != p.Target.Kind {
+				continue
+			}
+			if m.Key.Name != p.Target.Name {
+				continue
+			}
+			target = i
+			break
+		}
+		if target < 0 {
+			return nil, fmt.Errorf("no manifest matches the given patch: kind=%s, name=%s", p.Target.Kind, p.Target.Name)
+		}
+		patched, err := patcher(out[target], p)
+		if err != nil {
+			return nil, err
+		}
+		out[target] = *patched
+	}
+
+	return out, nil
+}
+
+func patchManifest(m provider.Manifest, patch config.K8sResourcePatch) (*provider.Manifest, error) {
+	if len(patch.YamlOps) == 0 {
+		return &m, nil
+	}
+
+	fullBytes, err := m.YamlBytes()
+	if err != nil {
+		return nil, err
+	}
+
+	process := func(bytes []byte) ([]byte, error) {
+		p, err := yamlprocessor.NewProcessor(bytes)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, o := range patch.YamlOps {
+			if o.Op != "replace" {
+				return nil, fmt.Errorf("%s operation is not supported currently", o.Op)
+			}
+			if err := p.ReplaceString(o.Path, o.Value); err != nil {
+				return nil, err
+			}
```

Comment:
nit: TODO comment helps us to track
```suggestion
			switch o.Op {
			case "replace":
				if err := p.ReplaceString(o.Path, o.Value); err != nil {
					return nil, err
				}
			case "add":
				// TODO: Support add operation for canary patch
				return nil, fmt.Errorf("%s operation is not supported currently", o.Op)
			case "remove":
				// TODO: Support remove operation for canary patch
				return nil, fmt.Errorf("%s operation is not supported currently", o.Op)
			default:
				return nil, fmt.Errorf("unexpected operation %s given", o.Op)
			}
```

Reply:
Nice.